### PR TITLE
Add `TripSettings` screen

### LINKED ISSUE #5: Implement View and Edit Trip Details screen
**Describe the solution you'd like**
- Navbar button click redirects to trip Details 
- offers possibility to read OR edit name, description, trip members

**Describe alternatives you've considered**

IMG_0364
IMG_0368
IMG_0373

### PR DESCRIPTION
**Pull request by DominicLuidold**
This PR implements the `TripSettings` screen as described in #5, including the [results of our discussion](https://github.com/vacatius/mobile-app/issues/5#issuecomment-833434370).

Here are a few screenshots that demonstrate the new functionality from the view of a user with the role `ADMIN`:

IMG_2438
IMG_2439

As a user with the role `MEMBER`, one can leave the group using a button underneath the member list.

Thanks to @whiteclouds7 who helped me with this PR (see #63).

Closes #5

### CODE REVIEW

**Comment by DominicLuidold** [comment]
@vacatius/mobile-app-team Currently only the trip "Project Dolphin 2.0" has two members. Please note when reviewing. Thanks!

**Comment by DominicLuidold** [comment]
@vacatius/mobile-app-team Please use this link to review the PR (the last commit only includes changes from `prettier -w /` after increasing the line length to `100`): https://github.com/vacatius/mobile-app/pull/70/files/d699cc83602c7fccca915fbeca8ae7351cd6a42f

**Comment by valentingc** [comment]
@DominicLuidold Can we pleae pretty-print the member roles? Something like first letter uppercase and then lowercase ("Admin", "Member")? Then it wouldn't look so technical. Maybe can also leave it out for regular members and just show it for admins?

File: src/screens/TripSettings/TripSettings.tsx
Comment (valentingc) [inline src/screens/TripSettings/TripSettings.tsx]:
```suggestion
            <KeyboardAwareScrollView keyboardShouldPersistTaps="handled">
```

File: src/routes/TripTabs.tsx
Comment (valentingc) [inline src/routes/TripTabs.tsx]:
```suggestion
```

File: src/screens/TripSettings/TripSettings.tsx
Comment (valentingc) [inline src/screens/TripSettings/TripSettings.tsx]:
```suggestion
                            {member.role === TripUserRole.Admin && (
```
Attention - you need to add the import as well, so accepting it here is not enough.

File: src/services/i18n.ts
Comment (valentingc) [inline src/services/i18n.ts:152]:
```suggestion
                        dialogTitle: "Really remove member?",
```

**Review by valentingc — CHANGES_REQUESTED** [review]
Good work. Found some minor things only.

File: src/screens/TripSettings/TripSettings.tsx
Comment (whiteclouds7) [inline src/screens/TripSettings/TripSettings.tsx]: @vacatius/mobile-app-team maybe we should discuss whether a member is only removed after clicking the update button or already after the admin has pressed the paperbin icon. Because if we decide on only removing them after presssing the update button, we probalby want to move it to the bottom of the screen, otherwise i think it is fine. But it could use a better tranlsation, something along the lines of "update trip information"? otherwise I think it could be a bit unclear

File: src/screens/TripSettings/TripSettings.tsx
Comment (whiteclouds7) [inline src/screens/TripSettings/TripSettings.tsx]:
```suggestion
                {data.trip.members.map((member) => (
                    <ListItem key={member.user.id} bottomDivider>
```
According to the [react docs](https://reactjs.org/docs/lists-and-keys.html#keys) they advise you to use unique identifiers if you have them before using the index which if I'm right the user id is

File: src/screens/TripSettings/TripSettings.tsx
Comment (whiteclouds7) [inline src/screens/TripSettings/TripSettings.tsx:303]:
```suggestion
                {data.trip.admin.id !== currentUser?.id && data.trip.members.length > 1 && settingsMode === Mode.EDIT && (
```
depends again if a member can only leave when being in edit mode or not

File: src/services/i18n.ts
Comment (whiteclouds7) [reply src/services/i18n.ts:152]: @valentingc We probably have to go over the translations anyway, because I think we're pretty inconsistent with starting words with a capital letter or not, e.g. Really leave **T**rip?

**Review by whiteclouds7 — CHANGES_REQUESTED** [review]
I've added two comments about when you can leave a trip which we probably have to discuss in the team. Apart from that great job @DominicLuidold 😄

**Review by valentingc — APPROVED** [review]

File: src/services/i18n.ts
Comment (DominicLuidold) [reply src/services/i18n.ts:152]: I've created #73 for resolving this issue as this is out of scope for this PR.

**Comment by DominicLuidold** [comment]
PR is on hold until #74 gets merged.

**Comment by DominicLuidold** [comment]
We've decided on leaving the current implementation as it is and only update the button text to make it more clear what's happening.

**Review by whiteclouds7 — APPROVED** [review]

**Pull request merged**
